tests/manage/z_cluster/test_coredump_check_for_ceph_daemon_crash.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import pytest
+
+from ocs_ci.framework.testlib import ManageTest, tier1, bugzilla
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.helpers.disruption_helpers import Disruptions
+from ocs_ci.helpers.helpers import run_cmd_verify_cli_output
+from ocs_ci.ocs.resources.pod import (
+    get_pod_node,
+    get_mon_pods,
+    get_ceph_tools_pod,
+    get_mgr_pods,
+    get_osd_pods,
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@bugzilla("1904917")
+class TestKillCephDaemon(ManageTest):
+    """
+    Verify coredump getting generated for ceph daemon crash
+
+    """
+
+    @pytest.fixture(scope="function", autouse=True)
+    def teardown(self, request):
+        """
+        Archive a crash report so that it is no longer considered for the
+        RECENT_CRASH health check and does not appear in the
+        crash ls-new output
+
+        """
+
+        def finalizer():
+            tool_pod = get_ceph_tools_pod()
+            tool_pod.exec_ceph_cmd(ceph_cmd="ceph crash archive-all", format=None)
+
+        request.addfinalizer(finalizer)
+
+    @pytest.mark.parametrize(
+        argnames=["daemon_type"],
+        argvalues=[
+            pytest.param(*["mon"], marks=pytest.mark.polarion_id("OCS-2491")),
+            pytest.param(*["osd"], marks=pytest.mark.polarion_id("OCS-2492")),
+            pytest.param(*["mgr"], marks=pytest.mark.polarion_id("OCS-2493")),
+        ],
+    )
+    def test_coredump_check_for_ceph_daemon_crash(self, daemon_type):
+        """
+        Verify coredumpctl list updated after killing daemon
+
+        """
+        log.info(f"Get Node name where {daemon_type} pod running")
+        if daemon_type == "mon":
+            mon_pod_nodes = [get_pod_node(pod) for pod in get_mon_pods()]
+            node_obj = mon_pod_nodes[0]
+        elif daemon_type == "mgr":
+            mgr_pod_nodes = [get_pod_node(pod) for pod in get_mgr_pods()]
+            node_obj = mgr_pod_nodes[0]
+        elif daemon_type == "osd":
+            osd_pod_nodes = [get_pod_node(pod) for pod in get_osd_pods()]
+            node_obj = osd_pod_nodes[0]
+        node_name = node_obj.name
+
+        log.info(
+            "Delete 'posted' directory /var/lib/rook/openshift-storage/crash/posted/"
```

Comment:
Better to change to " Delete the contents of 'posted' directory ...." as you are removing all the files in the dir "posted" instead of the directory "posted" itself.